rasterio/env.py

```diff
@@ -104,8 +106,7 @@ def __init__(self, aws_session=None, aws_access_key_id=None,
             self.aws_session._session.get_credentials()
             if self.aws_session else None)
         self.options = options.copy()
-        self.previous_options = {}
-        defenv()
+        self.context_options = {}
```

Comment:
Renamed `previous_options` to `context_options` as this is the name for config options defined outside (not before) the current `Env` instance.